app/lib/dates.lib.js

```diff
@@ -9,6 +9,27 @@ const february = 2
 const lastDayOfFebruary = 28
 const lastDayOfFebruaryLeapYear = 29
 
+/**
+ * Given a return version and return cycle end date, return the earliest date as a string in the ISO format.
+ *
+ * @param {Array} returnVersion - The list of dates to check.
+ * @returns {String} - The earliest date as a string in the ISO format.
+ * @private
+ */
+function earliestDate(dates) {
```

Comment:
Can I suggest a tweak to the name, to make it a bit more descriptive?

```suggestion
function determineEarliestDate(dates) {
```

Looking at `app/lib/general.lib.js` probably gives a better feel/example of the naming of helper functions like this.